src/Umbraco.Web.UI.Client/src/common/services/navigation.service.js

```diff
@@ -288,9 +288,12 @@ function navigationService($rootScope, $routeParams, $log, $location, $q, $timeo
                 throw "args.tree cannot be null";
             }
 
-            if (mainTreeEventHandler) {
-                //returns a promise
-                return mainTreeEventHandler.syncTree(args);
+            if (mainTreeEventHandler) {                
+                //returns a promise, but wrap it in a time out,
+                // on some occasions the syncTree method is not defined yet in the umb - tree directive(U4 - 11167)
+                $timeout(function () {                   
```

Comment:
this changes how the logic flows. Before we are returning the result of `mainTreeEventHandler.syncTree(args);` but now it is returning nothing. Returning from within the $timeout is not the same thing.

We are already checking if `mainTreeEventHandler` exists, why not also check for `mainTreeEventHandler.syncTree`, perhaps this check is all that is needed

Reply:
Hi Shannon,

That was my first solution. But the side effect was that the tree never selected the current node as active. If this problem occurs in 4th level content item, the content tree remains collapsed.

That's why I used the timeout. Maybe only execute the timeout when syncTree is not defined ?

Reply:
I updated it to only use a timeout when the syncTree method is not defined